superset/views/core.py

```diff
@@ -1111,139 +1111,91 @@ def save_or_overwrite_slice(
     @api
     @has_access_api
     @event_logger.log_this
-    @expose("/tables/<int:db_id>/<schema>/<substr>/")
-    @expose("/tables/<int:db_id>/<schema>/<substr>/<force_refresh>/")
-    @expose("/tables/<int:db_id>/<schema>/<substr>/<force_refresh>/<exact_match>")
-    def tables(  # pylint: disable=too-many-locals,no-self-use,too-many-arguments
+    @expose("/tables/<int:db_id>/<schema>/")
+    @expose("/tables/<int:db_id>/<schema>/<force_refresh>/")
```

Comment:
Since we are here, can we change `force_refresh` to a search params, too?

Reply:
Could this be the cause of #21476?

in conjunction with [tables.ts](superset-frontend/src/hooks/apiResources/tables.ts):
```javascript
export function fetchTables({
  dbId,
  schema,
  forceRefresh,
  keyword,
}: FetchTablesQueryParams) {
  const encodedSchema = schema ? encodeURIComponent(schema) : '';
  const encodedKeyword = keyword ? encodeURIComponent(keyword) : 'undefined';
  // TODO: Would be nice to add pagination in a follow-up. Needs endpoint changes.
  const endpoint = `/superset/tables/${
    dbId ?? 'undefined'
  }/${encodedSchema}/${encodedKeyword}/${forceRefresh}/`;
  return SupersetClient.get({ endpoint }) as Promise<QueryData>;
}
```